uaclient/contract.py

```diff
@@ -177,6 +179,41 @@ def report_machine_activity(self):
             machine_token["activityInfo"] = response
             self.cfg.write_cache("machine-token", machine_token)
 
+    def request_magic_token(self, magic_id: str):
+        """Request magic token for a user magic id.
+
+        This token should be used to allow UA to perform
+        an attach operation on the machine.
+        """
+        headers = self.headers()
+        headers.update({"Authorization": "Bearer {}".format(magic_id)})
+        url = API_V1_MAGIC_ATTACH
+
+        try:
+            response, _ = self.request_url(url, headers=headers)
+        except exceptions.UrlError as e:
+            if isinstance(e, exceptions.ContractAPIError):
+                if hasattr(e, "code"):
+                    if e.code == 400:
+                        raise exceptions.MagicAttachExpiredID(
+                            magic_id=magic_id
+                        )
+                    elif e.code == 401:
+                        raise exceptions.MagicAttachInvalidID(
+                            magic_id=magic_id
+                        )
+                    elif e.code == 404:
+                        raise exceptions.MagicAttachIDNotResolved(
+                            magic_id=magic_id
+                        )
+                raise e
+            with util.disable_log_to_console():
```

Comment:
unrelated out-of-scope comment: Do we ever want logging to go to the console? Maybe we can remove the console logger entirely and use print/event.info the few times we do want to print to the console.